### PR TITLE
 feat(*): project list and detail

### PR DESCRIPTION
**Pull request by qiuxiaolian**
**What this PR does / why we need it**:

- 项目列表
screen shot 2018-12-21 at 9 31 02 am
- 项目详情
screen shot 2018-12-21 at 9 48 44 am
- 新增项目
screen shot 2018-12-21 at 9 55 07 am

- EllipsisMenu component
screen shot 2018-12-21 at 9 48 37 am
- Detail component
- eslint + perttier + precommit for  consistent code

**Which issue(s) this PR is related to** *(optional, link to 3rd issue(s))*:

Fixes #

Reference to #

**Special notes for your reviewer**:

/cc @caosm @cd1989 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: web/src/components/project/component/list.js [hunk 1]
```diff
@@ -1,10 +1,107 @@
 import React from 'react';
```
Comment (artecher) [inline web/src/components/project/component/list.js]: About naming, please refer to `https://github.com/airbnb/javascript/tree/master/react#naming`. I suggest the initial letter for the component name be uppercase.

**Comment by cd1989** [comment]
lgtm

File: web/src/components/project/component/detail/index.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import React from 'react';
```
Comment (artecher) [inline web/src/components/project/component/detail/index.js]: Please try not to implement the actual logic in the `index.js`. Move the logic to other file and keep the index as just for exporting.

Comment (qiuxiaolian) [reply web/src/components/project/component/detail/index.js]: 👌

File: web/src/components/project/component/list.js (hunk 1 above)
Comment (cd1989) [reply web/src/components/project/component/list.js]: I will add a CONTRIBUTING file to cover such coding conventions both for frontend and backend. And investigate tools that can help auto check.

File: web/package.json [hunk 3]
```diff
@@ -16,6 +16,8 @@
     "raf": "3.4.0"
   },
   "scripts": {
+    "precommit": "lint-staged",
+    "lint": "eslint src",
     "start": "REACT_APP_BACKEND_SERVICE=http://192.168.19.96:30020/api/v1 node scripts/start.js",
```
Comment (supereagle) [inline web/package.json]: Hard code `192.168.19.96:30020`?

File: web/src/locale/zh-CN.json [hunk 4]
```diff
@@ -17,7 +19,14 @@
   },
   "workflow": {
     "recentVersion": "最近版本",
-    "creator": "创建者",
-    "creationTime": "创建时间"
+    "creator": "创建者"
+  },
+  "project": {
+    "workflowCount": "流水线数量",
```
Comment (supereagle) [inline web/src/locale/zh-CN.json]: Workflow is `工作流`, not `流水线`.

File: web/package.json (hunk 3 above)
Comment (qiuxiaolian) [reply web/package.json]: it's for test, can be removed when the api-proxy is ready

**Pull request merged**